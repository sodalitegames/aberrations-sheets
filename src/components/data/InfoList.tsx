import React from 'react';
import NewlineText from '../NewlineText';
import Tooltip from '../Tooltip';

export interface InfoListItem {
  tooltip?: string[];
  value: string;
  clamp?: boolean;
}

interface InfoListProps {
  list: string[] | InfoListItem[];
}

const InfoList: React.FC<InfoListProps> = ({ list }) => {
  return (
    <React.Fragment>
      {list.map((item, index: number) => {
        if (!item) {
          return (
            <span key={index} className="mt-1 text-sm text-gray-600">
              undefined (info list error)
            </span>
          );
        }

        // Check if item is a string
        if (typeof item === 'string') {
          return (
            <span key={index} className="mt-1 text-sm text-gray-600">
              <NewlineText text={String(item || '')} />
            </span>
          );
        }

        return (
          <Tooltip key={index} message={item.tooltip || []}>
            <span className="mt-1 text-sm text-gray-600 line-clamp-2">
              <NewlineText text={String(item.value || '')} />
            </span>
          </Tooltip>
        );
      })}
    </React.Fragment>
  );
};

export default InfoList;
