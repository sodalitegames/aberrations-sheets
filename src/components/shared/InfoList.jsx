import React from 'react';

const InfoList = ({ list }) => {
  return (
    <React.Fragment>
      {list.map((item, index) => {
        // check if it has a title
        if (item.title) {
          return (
            <p key={index} title={item.title} className="mt-1 text-sm text-gray-600 truncate">
              {item.value}
            </p>
          );
        }

        return (
          <p key={index} className="mt-1 text-sm text-gray-600">
            {item}
          </p>
        );
      })}
    </React.Fragment>
  );
};

export default InfoList;
