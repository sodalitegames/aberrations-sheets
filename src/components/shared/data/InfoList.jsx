import React from 'react';
import NewlineText from '../../utility/NewlineText';

const InfoList = ({ list }) => {
  return (
    <React.Fragment>
      {list.map((item, index) => {
        // check if it has a title
        if (item.title) {
          return (
            <span key={index} title={item.title} className="mt-1 text-sm text-gray-600 line-clamp-2">
              <NewlineText text={String(item.value)} />
            </span>
          );
        }

        return (
          <span key={index} className="mt-1 text-sm text-gray-600">
            <NewlineText text={String(item)} />
          </span>
        );
      })}
    </React.Fragment>
  );
};

export default InfoList;
