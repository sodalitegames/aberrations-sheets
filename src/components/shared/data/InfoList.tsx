import React from 'react';
import NewlineText from '../../utility/NewlineText';

export interface InfoListItem {
  title?: string;
  value: string;
  clamp?: boolean;
}

interface InfoListProps {
  list: string[] | InfoListItem[];
}

const InfoList: React.FC<InfoListProps> = ({ list }) => {
  return (
    <React.Fragment>
      {list.map((item, index: number) => {
        if (!item) {
          return (
            <span key={index} className="mt-1 text-sm text-gray-600">
              undefined (info list error)
            </span>
          );
        }

        // Check if item is a string
        if (typeof item === 'string') {
          return (
            <span key={index} className="mt-1 text-sm text-gray-600">
              <NewlineText text={String(item || '')} />
            </span>
          );
        }

        return (
          <span key={index} title={item.title || ''} className="mt-1 text-sm text-gray-600 line-clamp-2">
            <NewlineText text={String(item.value || '')} />
          </span>
        );
      })}
    </React.Fragment>
  );
};

export default InfoList;
