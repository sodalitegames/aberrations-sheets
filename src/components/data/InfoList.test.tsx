import { render } from '@testing-library/react';
import InfoList, { InfoListItem } from './InfoList';

it('renders the InfoList component with a string array', () => {
  const mockInfo = ['Hello there', 'this is some mock info'];
  expect(render(<InfoList list={mockInfo} />)).toMatchSnapshot();
});

it('renders the InfoList component with an InfoListItem array', () => {
  const mockInfo: InfoListItem[] = [
    { value: 'Hello there' },
    { tooltip: ['this is some mock info but longer'], value: 'this is some mock info' },
    { value: 'this is some really long mock info that is so long that it would need to be line clamped', clamp: true },
  ];
  expect(render(<InfoList list={mockInfo} />)).toMatchSnapshot();
});
