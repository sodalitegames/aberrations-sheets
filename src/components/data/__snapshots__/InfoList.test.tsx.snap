// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the InfoList component with a string array 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <span
        class="mt-1 text-sm text-gray-600"
      >
        <span
          class="space-y-1"
        >
          <p>
            Hello there
          </p>
        </span>
      </span>
      <span
        class="mt-1 text-sm text-gray-600"
      >
        <span
          class="space-y-1"
        >
          <p>
            this is some mock info
          </p>
        </span>
      </span>
    </div>
  </body>,
  "container": <div>
    <span
      class="mt-1 text-sm text-gray-600"
    >
      <span
        class="space-y-1"
      >
        <p>
          Hello there
        </p>
      </span>
    </span>
    <span
      class="mt-1 text-sm text-gray-600"
    >
      <span
        class="space-y-1"
      >
        <p>
          this is some mock info
        </p>
      </span>
    </span>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`renders the InfoList component with an InfoListItem array 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <span
        class="mt-1 text-sm text-gray-600 line-clamp-2"
        title=""
      >
        <span
          class="space-y-1"
        >
          <p>
            Hello there
          </p>
        </span>
      </span>
      <span
        class="mt-1 text-sm text-gray-600 line-clamp-2"
        title="this is some mock info but longer"
      >
        <span
          class="space-y-1"
        >
          <p>
            this is some mock info
          </p>
        </span>
      </span>
      <span
        class="mt-1 text-sm text-gray-600 line-clamp-2"
        title=""
      >
        <span
          class="space-y-1"
        >
          <p>
            this is some really long mock info that is so long that it would need to be line clamped
          </p>
        </span>
      </span>
    </div>
  </body>,
  "container": <div>
    <span
      class="mt-1 text-sm text-gray-600 line-clamp-2"
      title=""
    >
      <span
        class="space-y-1"
      >
        <p>
          Hello there
        </p>
      </span>
    </span>
    <span
      class="mt-1 text-sm text-gray-600 line-clamp-2"
      title="this is some mock info but longer"
    >
      <span
        class="space-y-1"
      >
        <p>
          this is some mock info
        </p>
      </span>
    </span>
    <span
      class="mt-1 text-sm text-gray-600 line-clamp-2"
      title=""
    >
      <span
        class="space-y-1"
      >
        <p>
          this is some really long mock info that is so long that it would need to be line clamped
        </p>
      </span>
    </span>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
